deploy-queue/tests/blocking_deployments_tests.rs

```diff
@@ -0,0 +1,258 @@
+// This file contains focused tests for the blocking_deployments.sql query, with each test
+// being self-contained and inserting only the specific data needed for its scenario.
+// 
+// Each test can run in parallel since they use isolated database connections.
+
+use anyhow::Result;
+
+#[path = "common/test_db_setup.rs"]
+mod database_helpers;
+
+#[path = "fixtures/deployment.rs"]
+mod deployment_fixtures;
+
+extern crate deploy_queue;
+
+use sqlx::{Pool, Postgres, Row};
+
+/// Assert that a deployment is blocked by exactly the expected deployments
+/// 
+/// # Arguments
+/// * `deployment_id` - The deployment to check for blocking
+/// * `expected_blockers` - List of deployment IDs that should be blocking this deployment (empty = not blocked)
+async fn assert_blocking_deployments(
+    pool: &Pool<Postgres>, 
+    deployment_id: i64, 
+    expected_blockers: Vec<i64>
+) -> Result<()> {
+    // Run the blocking deployments query
+    let sql = include_str!("../queries/blocking_deployments.sql");
+    let blocker_rows = sqlx::query(sql)
+        .bind(deployment_id)
+        .fetch_all(pool)
+        .await?;
+    
+    // Extract actual blocker IDs
+    let mut actual_blockers: Vec<i64> = blocker_rows
+        .iter()
+        .map(|row| row.get::<i64, _>("id"))
+        .collect();
+    
+    // Sort both vectors for consistent comparison
+    actual_blockers.sort();
+    let mut expected_sorted = expected_blockers.clone();
+    expected_sorted.sort();
+    
+    // Assert they match
+    assert_eq!(
+        actual_blockers, 
+        expected_sorted,
+        "Deployment {} blocking mismatch:\n  Expected blockers: {:?}\n  Actual blockers: {:?}",
+        deployment_id,
+        expected_sorted,
+        actual_blockers
+    );
+    
+    Ok(())
+}
+
+// Scenario 1: Simple blocking case
+// us-west-2 region: deployment 1 should block deployment 2
+#[tokio::test]
+async fn test_blocked_by_running_component_same_region() -> Result<()> {
+    let pool = database_helpers::setup_test_db().await?;
+    sqlx::query!(
+        "INSERT INTO deployments (id, region, component, environment, version, url, note, start_timestamp, finish_timestamp, cancellation_timestamp, concurrency_key) 
+         VALUES 
+             ($1, $2, $3, $4, $5, $6, $7, NOW() - INTERVAL '5 minutes', NULL, NULL, NULL),
+             ($8, $9, $10, $11, $12, $13, $14, NULL, NULL, NULL, NULL)",
+        1001, "us-west-2", "api-server", "prod", "v1.0.0", "https://github.com/api-server/v1.0.0", "Running deployment",
+        1002, "us-west-2", "web-frontend", "prod", "v2.1.0", "https://github.com/web-frontend/v2.1.0", "Queued deployment"
+    ).execute(&pool).await?;
```

Comment:
I'd move the values into the sql statement instead of using `$1` to `$14`, I think that'd be more readable. Same goes for the other tests below.